packages/@n8n/nodes-langchain/utils/helpers.ts

```diff
@@ -10,6 +10,21 @@ import type { BaseOutputParser } from '@langchain/core/output_parsers';
 import type { BaseMessage } from '@langchain/core/messages';
 import { DynamicTool, type Tool } from '@langchain/core/tools';
 import type { BaseLLM } from '@langchain/core/language_models/llms';
+import type { BaseChatMemory } from 'langchain/memory';
+import type { BaseChatMessageHistory } from '@langchain/core/chat_history';
+
+function hasMethod<T>(obj: unknown, methodName: string | symbol): obj is T {
```

Comment:
It might be helpful when used on its own. But that's a bit silly since one could use `hasMethods` as it accepts arbitrary amount parameters. So I've merged it with `hasMethods` and removed the return type from `isBaseChatMemory` and `isBaseChatMessageHistory` as those would be correctly evaluated by passing generic to `hasMethods`